src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.Helpers.cs

```diff
@@ -39,9 +39,15 @@ private static JsonTypeInfo CreateTypeInfoCore(Type type, JsonConverter converte
         {
             JsonTypeInfo typeInfo = JsonTypeInfo.CreateJsonTypeInfo(type, converter, options);
             typeInfo.NumberHandling = GetNumberHandlingForType(typeInfo.Type);
+            JsonObjectCreationHandling? preferredCreationHandling = GetObjectCreationHandlingForType(typeInfo.Type);
             if (typeInfo.Kind == JsonTypeInfoKind.Object)
             {
                 typeInfo.UnmappedMemberHandling = GetUnmappedMemberHandling(typeInfo.Type);
+                typeInfo.PreferredPropertyObjectCreationHandling = preferredCreationHandling;
+            }
+            else if (preferredCreationHandling != null)
+            {
+                ThrowHelper.ThrowInvalidOperationException_ObjectCreationHandlingAttributeOnWrongJsonTypeInfoKind(type);
```

Comment:
Should make this check in the `JsonTypeInfo.Configure()`, so that users can always correct invalid attribute annotations.

Reply:
This one is wrong usage of attribute so IMO we should always throw (there is no combination here, it just never applies to specific property type)

Reply:
I agree that it's wrong usage of the attribute, however we've received customer feedback before that contract customization should allow users fix incorrectly applied attributes in types from third-party libraries.

Reply:
To be fair currently ThrowInvalidOperationException_JsonTypeInfoOperationNotPossibleForKind is thrown only from setters in JsonTypeInfo. I'll change code to match that. If we're changing that we should do it for all of the properties - I can create an issue as a follow up

Reply:
actually seems that extra logic in DefaultJsonTypeInfoResolver was not needed at all because I already do the validation in the setter